pkg/types/symboltable/symboltable.go

```diff
@@ -0,0 +1,419 @@
+package symboltable
+
+import (
+	"encoding/json"
+	"fmt"
+	gotypes "github.com/gofed/symbols-extractor/pkg/types"
+)
+
+type SymbolTable interface {
+	LookupSymbol(string) SymbolDef
+	//AddSymbol(string, SymbolDef) error
+	AddFunction(string, SymbolDef) error
+	AddVariable(string, SymbolDef) error
+	AddConstant(string, SymbolDef) error
+	AddDataType(string, SymbolDef) error
+}
+
+type DeclPos struct {
+	File string
+	Line uint
+}
+
+//NOTE: Def can be empty for variable, but not for the others;
+type SymbolDef struct {
+	Pos  DeclPos          `json:"-"`
+	Name string           `json:"name"`
+	Def  gotypes.DataType `json:"def,omitempty"`
+}
+
+
+func (sym *SymbolDef) GetPosition() DeclPos { return sym.Pos }
+func (sym *SymbolDef) GetName() string      { return sym.Name }
+func (sym *SymbolDef) GetType() string      { return sym.Def.GetType() }
+
+//TODO: maybe something like this would be part of types.go? Used just
+//      to find data type of processed symbol
+//TODO: signature... would be fine use same as json.Unmarshal
+func unmarshalDataType(b []byte) (gotypes.DataType, error) {
```

Comment:
This is already generated https://github.com/gofed/symbols-extractor/commit/6223b32617476e16b5dcb402af5513236cee0fc5